### DISCUSSION TITLE
How to save canvas as SVG from the API

### DISCUSSION THREAD

**Discussion by sphuber** [epam/ketcher#1583] [Q&A]
Thanks for developing Ketcher. I can successfully export a molecule using the graphical interface, however, I would like to do it in javascript. Can I export the canvas as SVG using the `ketcher` element?

**Comment by AndreiMazol** [comment]
Dear @sphuber ,

to generate image please use the following steps:

1. serialize struct to  string:

`const struct = await ketcher.getKet()`

2. generate Blob:

`const imgBlob = await ketcher.generateImage(struct)` - png

or 

`const imgBlob = await ketcher.generateImage(struct, *options)`

*options is an object with 'outputFormat' ('svg' or  'png') and 'backgroundColor' (optional) properties. for example:

`let options = {outputFormat: 'svg'}`

Regards,
Ketcher team

**Reply by 1443957906** [reply]
Are there any other properties in options besides outputFormat and backgroundColor?
For example, if I want to modify the bond width of the generated SVG?

**Comment by sphuber** [comment]
Thanks @AndreiMazol . In my version, it seems `getKet` is not defined. Is this a recent addition perhaps?
